core/tests/protractor_utils/ExplorationEditorPage.js

```diff
@@ -241,14 +259,22 @@ var ExplorationEditorPage = function() {
   };
 
   this.expectCannotSaveChanges = async function() {
+    await waitFor.visibilityOf(
+      saveChangesButton, 'Save changes button taking too long to appear');
```

Comment:
This is incorrect. The next statement expects that the saveChangesButton element is not present. Have you tried running e2e tests that use this method on your local machine? If not, I strongly recommend that you do so.

Reply:
Yes @kevintab95 I will do that.

Reply:
Yes, please make sure to test all your changes before creating a PR.

Reply:
Sure, I will do that. But I have a doubt regarding that. Is the issue of flakiness of the tests resolved? If not then how will I get to know if the error is because of the flakiness of the test or not. (I have never checked this so I don't know much about it)

Reply:
Tests generally flake on the CI. It should pass when you run on your local machine.

Reply:
Oh, is that so? Sure thing then, I will run e2e tests on my local machine first and then proceed. Thanks!